src/reducers/computeRenderedColumns.js

```diff
@@ -0,0 +1,232 @@
+  const columnCount = scrollableColumns.length;
+
+  if (availableScrollWidth === 0 || columnCount === 0) {
+    state.firstColumnIndex = 0;
+    state.endColumnIndex = 0;
+    state.firstColumnOffset = 0;
+    return {
+      endBufferCol: 0,
+      endViewportCol: 0,
+      firstBufferCol: 0,
+      firstViewportCol: 0,
+    };
+  }
+
+  // If our first or last index is greater than our columnCount,
+  // treat it as if the last column is at the end of the viewport
+  let { firstIndex, firstOffset, lastIndex } = columnAnchor;
+  if (firstIndex >= columnCount || lastIndex >= columnCount) {
+    lastIndex = columnCount - 1;
+  }
+
+  // Walk the viewport until filled with columns
+  // If lastIndex is set, walk backward so that column is the last in the viewport
+  let step = 1;
+  let startIdy = firstIndex;
+  let totalWidth = firstOffset;
+  if (lastIndex !== undefined) {
+    step = -1;
+    startIdy = lastIndex;
+    totalWidth = 0;
+  }
+
+  // Loop to walk the viewport until we've touched enough columns to fill its width
+  let columnIdx = startIdy;
+  let endIdy = columnIdx;
+  while (columnIdx < columnCount && columnIdx >= 0 &&
+      totalWidth < availableScrollWidth) {
+    totalWidth += updateColumnWidth(state, columnIdx);
+    endIdy = columnIdx;
+    columnIdx += step;
+  }
+
+  // Loop to walk the leading buffer
+  let firstViewportCol = Math.min(startIdy, endIdy);
+  const firstBufferCol = Math.max(firstViewportCol - bufferColumnCount, 0);
+  for (columnIdx = firstBufferCol; columnIdx < firstViewportCol; columnIdx++) {
+    updateColumnWidth(state, columnIdx);
+  }
+
+  // Loop to walk the trailing buffer
+  const endViewportCol = Math.max(startIdy, endIdy) + 1;
+  const endBufferCol = Math.min(endViewportCol + bufferColumnCount, columnCount);
+  for (columnIdx = endViewportCol; columnIdx < endBufferCol; columnIdx++) {
+    updateColumnWidth(state, columnIdx);
+  }
+
+  // Calculate offset needed to position column at the end of viewport
+  // This should be negative and represent how far the first column needs to be offscreen
+  if (lastIndex !== undefined) {
+    firstOffset = Math.min(availableScrollWidth - totalWidth, 0);
+  }
+
+  state.firstColumnIndex = firstViewportCol;
+  state.endColumnIndex = endViewportCol;
+  state.firstColumnOffset = firstOffset;
+
+  return {
+    endBufferCol,
+    endViewportCol,
+    firstBufferCol,
+    firstViewportCol,
+  };
+}
+
+/**
+ * Walk the columns to render and compute the width offsets and
+ * positions in the column buffer.
+ *
+ * NOTE (jordan) This alters state so it shouldn't be called
+ * without state having been cloned first.
+ *
+ * @param {!Object} state
+ * @param {{
+ *   endBufferCol: number,
+ *   endViewportCol: number,
+ *   firstBufferCol: number,
+ *   firstViewportCol: number,
+ * }} columnRange
+ * @param {boolean} viewportOnly
+ * @private
+ */
+function computeRenderedColumnOffsets(state, columnRange, viewportOnly) {
```

Comment:
similar to [computeRendererdColumnOffsets](https://github.com/schrodinger/fixed-data-table-2/blob/v1.0-beta/src/reducers/computeRenderedRows.js#L203)

**Quick description**
We step through each column in our column range (calculated above), and 
* store the column offsets here
* store the column buffer mapping